lib/ipmi_fru.c

```diff
@@ -3849,40 +3853,41 @@ static int ipmi_fru_upg_ekeying(struct ipmi_intf *intf, char *pFileName,
 	    != 0) {
 		lprintf(LOG_ERR, "Failed to get multirec size from file '%s'.",
 			pFileName);
-		return -1;
+		ERR_EXIT;
 	}
 	buf = malloc(fileMultiRecSize);
 	if (!buf) {
 		lprintf(LOG_ERR, "ipmitool: malloc failure");
-		return -1;
+		ERR_EXIT;
 	}
 	if (ipmi_fru_get_multirec_from_file(pFileName, buf, fileMultiRecSize,
 					    offFileMultiRec)
 	    != 0) {
 		lprintf(LOG_ERR, "Failed to get multirec from file '%s'.",
 			pFileName);
-		free(buf);
-		buf = NULL;
-		return -1;
+		ERR_EXIT;
 	}
 	if (ipmi_fru_get_adjust_size_from_buffer(buf, &fileMultiRecSize) != 0) {
 		lprintf(LOG_ERR, "Failed to adjust size from buffer.");
-		free(buf);
-		buf = NULL;
-		return -1;
+		ERR_EXIT;
 	}
 	if (write_fru_area(intf, &fruInfo, fruId, 0, offFruMultiRec,
 			   fileMultiRecSize, buf)
 	    != 0) {
 		lprintf(LOG_ERR, "Failed to write FRU area.");
+		ERR_EXIT;
+	}
+
+exit:
+	if (buf) {
```

Comment:
As you noted before, this check is superfluous. `free()` does nothing on NULL argument. That's standardized. It's safe to call `free(NULL)`. So let's put this `free()` right before `return rc` and get rid of this check and the `buf = NULL`.

Reply:
It's now standardized, because for years and years it wasn't.  Are we sure someone compiling this isn't going to run into trouble with that?  In this case, they can reach buf with a null pointer as it's not a failure case in this specific method.

Reply:
Also, in a previous PR you indicated I needed to leave in the setting of NULL to a pointer, and this included cases where that had no effect, such as at the end of a method -- maybe you meant to exclude this case.

Reply:
I thought I had removed anything I added, as all free calls were changed to free_n()